Content.Client/Audio/AmbientSoundSystem.cs

```diff
@@ -164,36 +175,37 @@ private void ClearSounds()
         {
-            //TODO: Make this produce a hashset of nearby entities again.
-            var sourceDict = new Dictionary<string, List<AmbientSoundComponent>>(16);
+            var sourceDict = new Dictionary<string, SortedList<float, AmbientSoundComponent>>(16);
             var ambientQuery = GetEntityQuery<AmbientSoundComponent>();
-            var xformQuery = GetEntityQuery<TransformComponent>();
 
-            foreach (var entity in _lookup.GetEntitiesInRange(coordinates, _maxAmbientRange + RangeBuffer))
+            // TODO add variant of GetComponentsInRange that also returns the transform component.
+            foreach (var entity in _lookup.GetEntitiesInRange(coords, _maxAmbientRange, flags: _flags))
             {
-                if (!ambientQuery.TryGetComponent(entity, out var ambientComp) ||
-                    !ambientComp.Enabled ||
-                    !xformQuery.GetComponent(entity).Coordinates.TryDistance(EntityManager, coordinates, out var range) ||
-                    range > ambientComp.Range)
-                {
+                if (!ambientQuery.TryGetComponent(entity, out var ambientComp) || !ambientComp.Enabled)
                     continue;
-                }
 
-                var key = _audio.GetSound(ambientComp.Sound);
+                var xform = xformQuery.GetComponent(entity);
+                var delta = xform.ParentUid == playerXform.ParentUid
+                    ? xform.LocalPosition - playerXform.LocalPosition
+                    : xform.WorldPosition - coords.Position;
 
-                if (!sourceDict.ContainsKey(key))
-                    sourceDict[key] = new List<AmbientSoundComponent>(MaxSingleSound);
+                var range = delta.Length;
+                if (range >= ambientComp.Range)
+                    continue;
 
-                sourceDict[key].Add(ambientComp);
-            }
+                string key;
 
-            // TODO: Just store the distance from above...
-            foreach (var (key, val) in sourceDict)
-            {
-                sourceDict[key] = val.OrderByDescending(x =>
-                    Transform(x.Owner).Coordinates.TryDistance(EntityManager, coordinates, out var dist) ? dist * (x.Volume + 32) : float.MaxValue).ToList();
+                if (ambientComp.Sound is SoundPathSpecifier path)
+                    key = path.Path?.ToString() ?? string.Empty;
+                else
+                    key = ((SoundCollectionSpecifier) ambientComp.Sound).Collection ?? string.Empty;
+
+                var list = sourceDict.GetOrNew(key);
+
+                // wtf is this sorting even for? If its for prioritizing higher playing more audible sounds before others, shouldn't further away be counteracting higher volume?
+                list.Add(range * (ambientComp.Volume + 32), ambientComp);
```

Comment:
I'm not sure what the logic behind the previous sorting was, or where the magic "+32" comes from.

Reply:
also yeah there's a typo in the comment, but the comment should get resolved before merging anyways

Reply:
i don't remember what the +32 is for but the idea is indeed to prioritize by volume. a further away but louder sound gets higher priority.

Reply:
Okay, I've just left it as is, though I did flip the sorting so it matches how it is currently sorted on master.